AllEvents.TicketManagement/src/Core/AllEvents.TicketManagement.Application/Contracts/IEventRepository.cs

```diff
@@ -1,12 +1,14 @@
-﻿using AllEvents.TicketManagement.Application.Models;
-using AllEvents.TicketManagement.Domain.Entities;
+﻿using AllEvents.TicketManagement.Domain.Entities;
 
 namespace AllEvents.TicketManagement.Application.Contracts
 {
     public interface IEventRepository
     {
+        Task<List<Event>> GetPagedEventsAsync(int pageIndex, int pageSize, bool includeDeleted = false);
         Task<int> GetCountAsync();
-        Task<List<Event>> GetPagedEventsAsync(int page, int pageSize);
+        Task<bool> ExistsAsync(Guid eventId);
+        Task SoftDeleteAsync(Guid eventId);
```

Comment:
@KKDinev20  
Add the method on the entity 
Event.Delete()
Event.Restore()
and finally in the command you will have saveChanges
such operation should not be responsibility of the Repository 
the Event should know what to do 
you don't need it on the repository